rust/lancedb/src/index/vector.rs

```diff
@@ -46,17 +47,19 @@ impl VectorIndex {
     }
 }
 
+#[skip_serializing_none]
 #[derive(Debug, Deserialize)]
 pub struct VectorIndexMetadata {
-    pub metric_type: String,
-    pub index_type: String,
+    pub metric_type: Option<String>,
+    pub index_type: Option<String>,
 }
 
+#[skip_serializing_none]
 #[derive(Debug, Deserialize)]
 pub struct VectorIndexStatistics {
```

Comment:
Also here, are we carrying ScalarIndexStats from `VectorIndexStats`?

Reply:
will change the name.